### PR TITLE
web app: Updates to az webapp up command - removes preview tag & other changes

### LINKED ISSUE #8203: az webapp up to warn user based on number of apps deployed to the same service plan
**Is your feature request related to a problem? Please describe.**
Today we use default rg and service plan for up command, people share the same subscription will deploy their apps to the same service plan (worker) in a given region. There are performance impacts when you deploy more than 5 apps to the same service plan.

**Describe the solution you'd like**
In up command, check the number of existing apps in the default service plan, if the # is 5 or larger, display the following text in the console output: There are <n> sites have been deployed to the same hosting VM for this Azure region, to prevent performance impact please delete existing site(s) or switch to a different default resource group using "az configure" command (https://go.microsoft.com/fwlink/?linkid=2056936).

### PR DESCRIPTION
**Pull request by panchagnula**
Fixes #8203 , #8665 , #8287
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by panchagnula** [comment]
@yugangw-msft , given today is the deadline for code complete can you review this today please?

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -284,6 +285,7 @@ def enable_zip_deploy(cmd, resource_group_name, name, src, timeout=None, slot=No
     # Read file content
     with open(os.path.realpath(os.path.expanduser(src)), 'rb') as fs:
         zip_content = fs.read()
+        logger.warning("Starting Zip Deployment")
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: Use lower case such as `Starting zip deployment`? This is to be consistent with the other occasions mentioning the same term.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -2261,30 +2263,47 @@ def create_deploy_webapp(cmd, name, location=None, sku=None, dryrun=False):  # p
         client.app_service_plans.create_or_update(rg_name, asp, plan_def)
         logger.warning("App service plan creation complete")
         _create_new_app = True
+        _show_tooMany_apps_warn = False
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: `_show_too_many_apps_warn`?

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -2261,30 +2263,47 @@ def create_deploy_webapp(cmd, name, location=None, sku=None, dryrun=False):  # p
...
+            logger.warning('Updating runtime version from %s to %s',
+                           site_config.linux_fx_version, runtime_version)
+            update_site_configs(cmd, rg_name, name, linux_fx_version=runtime_version)
+        elif os_val == 'Windows' and site_config.windows_fx_version != runtime_version:
+            logger.warning('Updating runtime version from %s to %s',
+                           site_config.windows_fx_version, runtime_version)
+            update_site_configs(cmd, rg_name, name, windows_fx_version=runtime_version)
+
         if do_deployment and not is_skip_build:
+            _set_build_appSetting = True
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: `_set_build_app_setting `

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_create_util.py [hunk 4]
```diff
@@ -193,17 +198,20 @@ def parse_netcore_version(file_path):
...
-            data.append(c[:3])
-        version_detected = sorted(data, key=float, reverse=True)
+        data = json.load(data_file)
+        for key, value in data.items():
+            if key == 'engines' and 'node' in value:
+                value_detected = value['node']
+                non_decimal = re.compile(r'[^\d.]+')
+                # remove the string ~ or  > that sometimes exists in version value
+                c = non_decimal.sub('', value_detected)
+                # reduce the version to '6.0' from '6.0.0'
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_create_util.py]: Either this comment or the code below is not right. `6.0` is only 3 characters long

Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_create_util.py]: Here is how i understand Python's slice notation 
c[:4] => this reads as :stop =>items from the beginning through stop-1 , so this is until character 3 so will change 6.0.0 to 6.0

Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_create_util.py]: checked below out. 
```bash
Python 2.7.5 (default, Jul 13 2018, 13:06:57)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-28)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> '6.0.0'[:4]
'6.0.'
>>> '6.0.0'[:3]
'6.0'
```

Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_create_util.py]: weird that's not what i am seeing i tried printing out the values on console

image

Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_create_util.py]: Also saw this article explaining it

https://stackoverflow.com/questions/509211/understanding-slice-notation

i am concerned though, that the behavior is going to be different with different Python versions.

Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_create_util.py]: Figured out the issue here with 10.0.0 vs 6.0.0 ,so changed this logic

**Comment by yugangw-msft** [comment]
@panchagnula, can you please remove the `package.json.txt`?

**Comment by yugangw-msft** [comment]
@adewaleo, FYI, the CI for this PR failed due to the flakiness of `test_vmss_run_command_e2e`. You can check out the trace if you want

**Pull request merged**